src/modules/roc_pipeline/converter.h

```diff
@@ -50,7 +51,9 @@ class Converter : public sndio::ISink, public core::NonCopyable<> {
     audio::NullWriter null_writer_;
 
     core::UniquePtr<audio::PoisonWriter> resampler_poisoner_;
-    core::UniquePtr<audio::ResamplerWriter> resampler_;
+    core::UniquePtr<audio::ResamplerWriter> resampler_writer_;
+    core::UniquePtr<audio::ResamplerMap> resampler_map_;
```

Comment:
Other maps (CodecMap, FormatMap) are passed from the upper code instead of creating them in the pipeline. Such design assumes that the user can somehow configure them map and then pass it to the pipeline, which will use the map to create instances of codecs/parsers/etc.

ResamplerMap currently does not have any configuration options, but I suggest to use the same approach for all maps for consistency.

Reply:
Or we can at least avoid allocating resampler map on heap and just use a local variable, so this:

```
        resampler_map_.reset(...)
        if (!resampler_map_) {
            return;
        }

        resampler_.reset(resampler_map_->new_resampler(...));
```

would become this:

```
        ResamplerMap resampler_map;

        resampler_.reset(resampler_map.new_resampler(...));
```